### PR TITLE
MLIBZ-2823 Correctly handle missing or null refresh token

### PR DESCRIPTION
**Pull request by b-stolyarov**
#### Description
Correctly handle missing or null refresh token

#### Changes
added checking for `refresh_token==null`

#### Tests
Instrumented

### CODE REVIEW

File: android-lib/src/main/java/com/kinvey/android/store/UserStore.java [hunk 1]
```diff
@@ -1116,7 +1116,9 @@ protected T executeAsync() throws IOException {
             UserStoreRequestManager requestManager = new UserStoreRequestManager(client, createBuilder(client));
             requestManager.setMICRedirectURI(redirectURI);
             GenericJson result = requestManager.getMICToken(token, clientId).execute();
-
+            if (result.get(ACCESS_TOKEN) == null || result.get(REFRESH_TOKEN) == null) {
```
Comment (vinaygahlawat) [inline android-lib/src/main/java/com/kinvey/android/store/UserStore.java]: It is allowed that you can have an `access_token` but not necessarily a `refresh_token`, so we cannot have this `||` logic here.

File: android-lib/src/main/java/com/kinvey/android/store/UserStore.java [hunk 2]
```diff
@@ -1150,6 +1152,9 @@ protected T executeAsync() throws IOException {
             UserStoreRequestManager requestManager = new UserStoreRequestManager(client, createBuilder(client));
             requestManager.setMICRedirectURI(redirectURI);
             GenericJson result = requestManager.getOAuthToken(clientId, username, password).execute();
+            if (result.get(ACCESS_TOKEN) == null || result.get(REFRESH_TOKEN) == null) {
```
Comment (vinaygahlawat) [inline android-lib/src/main/java/com/kinvey/android/store/UserStore.java]: Same comment as above.

File: android-lib/src/main/java/com/kinvey/android/store/UserStore.java (hunk 1 above)
Comment (b-stolyarov) [reply android-lib/src/main/java/com/kinvey/android/store/UserStore.java]: Fixed

File: android-lib/src/main/java/com/kinvey/android/store/UserStore.java (hunk 2 above)
Comment (b-stolyarov) [reply android-lib/src/main/java/com/kinvey/android/store/UserStore.java]: Fixed

**Pull request merged**